formats/label-to-vector-plugin/src/polus/plugins/formats/label_to_vector/dynamics/label_to_vector.py

```diff
@@ -0,0 +1,419 @@
+    roughly the TILE_SIZE**2 number of pixels.
+
+    Args:
+        image: A labeled image
+        slices: A list of slice tuples, one for each ROI.
+
+    Yields:
+        Returns a mask stack, a list of slice tuples for each roi, and the
+        centroid location.
+    """
+    # Set the maximum number of pixels to process at a time
+    max_pixels = constants.TILE_SIZE**2
+    current_label = 0
+    dims = image.ndim
+
+    # Outermost loop, resets values for next generation
+    while current_label < len(slices):
+        masks: list[numpy.ndarray] = []
+        extents_: list[numpy.ndarray] = []
+        total_pixels = 0  # Total number of pixels in current generation
+        max_slice_pixels = 0  # width*height of the largest ROI in the generation
+
+        # Stack a set of ROIs for the current generation
+        while total_pixels < max_pixels and current_label < len(slices):
+            # Create the mask and append to the list of masks
+            mask = image[slices[current_label]] == (current_label + 1)
+
+            # Check if the current ROI is larger than the largest ROI
+            current_size = mask.size
+            if current_size > max_slice_pixels:
+                # If the current ROI would give too many pixels, load it next iteration
+                if current_size * len(masks) > max_pixels:
+                    break
+
+                else:
+                    max_slice_pixels = mask.size
+
+            # Get the geometric median of the object
+            geo_median = geometric_median(mask)
+
+            # Calculate the extents of the region relative to the median
+            extent = numpy.zeros((2 * dims,))
+            extent[:dims] = -geo_median
+            extent[dims:] = numpy.asarray(mask.shape) - geo_median
+
+            masks.append(mask)
+            extents_.append(extent)
+
+            current_label += 1
+            total_pixels = max_slice_pixels * len(masks)
+
+        # Calculate the size of output needed to store all ROIs with the same centroid
+        extents = numpy.asarray(extents_).astype(int)
+        centroid: numpy.ndarray = -numpy.min(extents[:, :dims], axis=0) + 2
+        shape: numpy.ndarray = numpy.max(extents[:, dims:], axis=0) + centroid + 1
+
+        # Calculate offsets for each ROI to position the centroid at the same location
+        extents[:, :dims] += centroid
+        extents[:, dims:] += centroid
+
+        # Create the output, and stack the ROIs
+        stack = numpy.zeros(
+            [len(masks), *shape.astype(int).tolist()],
+            dtype=numpy.bool_,
+        )
+
+        sub_slices: list[tuple[slice, ...]] = []
+
+        for i, mask in enumerate(masks):
+            ext = extents[i]
+
+            slice_tuple = tuple(slice(ext[i], ext[i + dims]) for i in range(dims))
+
+            sub_slices.append(slice_tuple)
+
+            stack[(i, *slice_tuple)] = mask
+
+        yield stack, sub_slices, centroid
+
+
+def creeping_mean_filter(
```

Comment:
rename? this run the simulation, isn't the rest implementation details?

Reply:
We talked about it already, but I wonder if the simulation code could be written a bit more clearly but let's leave that to another iteration